ui/mirage/handlers/sync.js

```diff
@@ -27,6 +27,33 @@ export const associationsResponse = (schema, req) => {
   };
 };
 
+export const syncStatusResponse = (schema, req) => {
+  const { mount, name: secret_name } = req.params;
+  const records = schema.db.syncAssociations.where({ mount, secret_name });
+  if (!records.length) {
+    return new Response(404, {}, { errors: [] });
+  }
+  const STATUSES = ['SYNCING', 'SYNCED', 'UNSYNCED', 'UNSYNCING', 'INTERNAL_VAULT_ERROR'];
+  const generatedRecords = records.reduce((records, record, index) => {
+    const destinationType = record.type;
+    const destinationName = record.name;
+    record.sync_status = STATUSES[index];
+    const key = `${destinationType}/${destinationName}`;
+    records[key] = record;
+    return records;
+  }, {});
+  generatedRecords['aws-sm/my-aws-destination'] = {
+    ...generatedRecords['aws-sm/destination-aws'],
+    sync_status: 'UNKNOWN',
+    name: 'my-aws-destination',
+  };
+  return {
+    data: {
+      associated_destinations: generatedRecords,
+    },
+  };
+};
```

Comment:
Nice thanks for keep the mirage handler up to date. Is there a plan to use this elsewhere?

Reply:
Yes! I'm using it in tests